File: LeetCode/Pre/20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        dic = {"}":"{", ")":"(", "]":"["}
        for c in s:
            if c in dic.values():
                st.append(c)
            elif c in dic.keys():
                if st == [] or dic[c] != st.pop():
                    return False
        return st == []

File: LeetCode/Pre/20. Valid Parentheses.cpp
用stack进行对应，没啥难度
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for(char c:s){
            if(c=='(' || c=='[' || c=='{') st.push(c);
            else{
                if(st.empty()) return false;
                if(st.top()=='(' && c!=')') return false;
                if(st.top()=='[' && c!=']') return false;
                if(st.top()=='{' && c!='}') return false;
                st.pop();
            }
        }
        return st.empty();
    }
};
